ksqldb-serde/src/main/java/io/confluent/ksql/serde/protobuf/ProtobufSchemas.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.serde.protobuf;
+
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+
+public final class ProtobufSchemas {
+  private ProtobufSchemas() {
+  }
+
+  static Schema schemaWithName(final Schema schema, final String schemaName) {
+    if (schemaName == null) {
+      return schema;
+    }
+
+    final SchemaBuilder builder;
+    switch (schema.type()) {
+      case STRUCT:
+        builder = buildSchemaStruct(schema);
+        break;
+      case ARRAY:
+        builder = SchemaBuilder.array(schema.valueSchema());
+        break;
+      case MAP:
+        builder = SchemaBuilder.map(
+            schema.keySchema(),
+            schema.valueSchema()
+        );
+        break;
+      default:
+        builder = new SchemaBuilder(schema.type());
```

Comment:
Curious in what situation the root schema is not a `struct`? Should we just deal with `struct`?

Reply:
Done